POEM_056.md

```diff
@@ -0,0 +1,349 @@
+`jax` then the function definition would raise an exception because the output shape(s) would be
+unknown.  This is not an issue if `declare_partials` (see below)
+specifies that `jax` be used as the method to compute partial derivatives because in that case,
+there will already be a `jax` dependency.
+
+
+### Getting the metadata
+
+Variable metadata is retrieved from the callable object by passing that object to the `get_input_meta`
+and `get_output_meta` methods. Each function returns an iterator over (name, metadata_dict) tuples, 
+one for each input or output variable respectively.  For example, the following code snippet will
+print the name and shape of each output variable (assuming f is a wrapped function).
+
+```python
+for name, meta in f.get_output_meta():
+    print(name, meta['shape'])
+```
+
+Note that in addition to user supplied metadata, metadata for each added output will include a
+'deps' entry which contains a set of names of inputs that that output depends on.  This does
+not include any sparsity information if the input and/or output are array variables.  It only 
+shows whether a given output depends in any way upon a particular input variable.
+
+## Setting function default metadata
+
+Some metadata will be the same for all, or at least most of the variables within a given function,
+so we want to be able to specify those defaults easily without too much boilerplate.  That's the
+purpose of the `defaults` method.  For example:
+
+```python
+def func(a, b, c):
+    d = a * b * c
+    return d
+
+f = omf.wrap(func).defaults(shape=4, units='m')
+```
+
+Any metadata that is specific to a particular variable will override any defaults specified in
+`defaults`. For example:
+
+```python
+def func(a, b, c=np.ones(3)):  # shape of c is 3 which overrides the `defaults` shape
+    d = a * b
+    e = c * 1.5
+    return d, e
+
+f = omf.wrap(func).defaults(shape=4, units='m')
+```
+
+### Getting the function default metadata
+
+Currently there doesn't seem to be a need to add this, as the defaults are automatically applied
+to the variable metadata, but a function to do this could easily be added later if needed.
+
+
+### Assumed default values
+
+In order to stay consistent with OpenMDAO's default value policy, we'll assume the same default
+behavior for functions, so if no shape or default value is supplied for a function variable, we'll
+assume that is has the value 1.0.  If the `shape` is provided and either the default value is
+not provided or is provided as a scalar value, then the assumed default value will be
+`np.ones(shape) * scalar_default_value`, where `scalar_default_value` is 1.0 if not specified.
+If `shape` is provided along with a non-scalar default value that has a different shape, then
+an exception will be raised.
+
+
+## Setting non-default function metadata
+
+The `metadata` decorator can be used to specify metadata that is intended to apply to the function 
```

Comment:
`decorator` isn't actually used any more. Replace with `metadata method`

Reply:
all of the occurrences of 'decorator' should be gone now.